scripts/FigS4_t-cell-umap.R

```diff
@@ -0,0 +1,172 @@
+
+
+# load project
+renv::load()
+
+library(SingleCellExperiment)
+library(ggplot2)
+library(patchwork)
+
+# set default ggplot theme for UMAPs
+theme_set(
+  theme_classic() +
+    theme(
+      #plot.margin = margin(rep(20, 4)),
+      strip.background = element_rect(fill = "transparent", linewidth = 0.5),
+      # no axis ticks or labels
+      axis.line = element_blank(),
+      axis.ticks = element_blank(),
+      axis.text = element_blank(),
+      # add a square around each of the plots
+      panel.background = element_rect(colour = "black", linewidth = 0.5),
+      aspect.ratio = 1,
+      # remove boxes around legends 
+      legend.key = element_blank()
+    )
+)
+# Set up -----------------------------------------------------------------------
+
+sce_file <- here::here("s3_files", "SCPCS000049", "SCPCL000049_processed.rds")
+
+# t cell list
+# includes a list of all T cell types across all methods that are found in this sample
+t_cell_types <- readLines(here::here("sample-info", "SCPCS000049_T-cell-types.txt")) 
+
+# define output files
+tcell_umap_file <- here::here("figures", "pdfs", "FigS4A_tcell-umaps.pdf")
+
+# get cell types and umap embeddings for plotting
+umap_df <- readr::read_rds(sce_file) |>
+  # convert to data frame immediately for space
+  scuttle::makePerCellDF(
+    use.coldata = c(
+      "barcodes",
+      "consensus_celltype_annotation",
+      "singler_celltype_annotation",
+      "cellassign_celltype_annotation",
+      "scimilarity_celltype_annotation",
+      "infercnv_total_cnv"
+    ),
+    use.dimred = "UMAP"
+  ) |>
+  # remove rownames
+  tibble::as_tibble()
+
+# Faceted UMAP - panel A -------------------------------------------------------
+
+# list of cell type columns to lump and plot
+celltype_columns <- c(
+  "Consensus" = "consensus_celltype_annotation",
+  "SingleR" = "singler_celltype_annotation",
+  "CellAssign" = "cellassign_celltype_annotation",
+  "SCimilarity"= "scimilarity_celltype_annotation"
```

Comment:
Let's sync up here: for my pending heatmap, I was ordering these in the plot by alphabetical order (cellassign, scimilarity, singler). I think that makes more sense for the paper? This order is more how the sausage was made.